python/mlcroissant/mlcroissant/_src/operation_graph/operations/download.py

```diff
@@ -152,9 +153,11 @@ def _check_hash(self, filepath: epath.Path):
         with filepath.open("rb") as f:
             while chunk := f.read(_CHUNK_SIZE):
                 hash.update(chunk)
-        actual_hash = hash.hexdigest()
+        # The hash from the Croissant JSON-LD can be in either hex or base64, so check both
+        hex_hash = hash.hexdigest()
+        base64_hash = base64.b64encode(bytes.fromhex(hex_hash)).decode()
```

Comment:
Should you try to early return here by testing `if hex_hash != expected_hash` in order to spare the base64 encoding?

Reply:
I think you mean `if hex_hash == expected_hash` since `!=` would mean it didn't pass and we fallback to checking base64. But yeah, I can do that 👍